src/DotNetWorker/Definition/DefaultFunctionDefinition.cs

```diff
@@ -1,29 +1,32 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using Microsoft.Azure.Functions.Worker.Invocation;
 using Microsoft.Azure.Functions.Worker.OutputBindings;
 
 namespace Microsoft.Azure.Functions.Worker.Definition
 {
     internal class DefaultFunctionDefinition : FunctionDefinition
     {
-        public DefaultFunctionDefinition(FunctionMetadata metadata, IFunctionInvoker invoker, IEnumerable<FunctionParameter> parameters, OutputBindingsInfo outputBindings)
+        public const string InvokerKey = "AzureFunctions_Invoker";
+
+        private readonly Lazy<IDictionary<string, object>> _itemsLazy = new Lazy<IDictionary<string, object>>(() => new Dictionary<string, object>());
+
+        public DefaultFunctionDefinition(FunctionMetadata metadata, IEnumerable<FunctionParameter> parameters, OutputBindingsInfo outputBindings)
         {
             Metadata = metadata;
-            Invoker = invoker;
             Parameters = parameters.ToImmutableArray();
             OutputBindingsInfo = outputBindings;
         }
 
-        public override FunctionMetadata Metadata { get; set; }
+        public override FunctionMetadata Metadata { get; }
 
-        public override ImmutableArray<FunctionParameter> Parameters { get; set; }
+        public override ImmutableArray<FunctionParameter> Parameters { get; }
 
-        public override IFunctionInvoker Invoker { get; set; }
```

Comment:
The comments from the review said:
> Remove invoker. The middleware should create this if reached.

But creating the invoker requires the `MethodInfo` today. Without a major reshuffling, I created an `Items` dictionary here where we can store it without exposing the type. Thoughts? 

I could also refactor the IFunctionInvokerFactory so that it doesn't use a `MethodInfo` directly... I'll try that as well now and see what it looks like.

Reply:
We can have the more elaborate change with a later update. Let's land the initial set of changes first. We definitely want to remove all reflection types from the abstractions (goal should be to have 100% non-reflection paths), but that can come in a  later change.

Reply:
I didn't like the fact that I had to add `Items` (when we may not need it later) -- so I have another approach in my next commit.